l10n_it_delivery_note_inter_company/models/stock_delivery_note.py

```diff
@@ -27,13 +28,13 @@ def update_detail_lines(self):
             [("delivery_note_id", "!=", False)]
         )
```

Comment:
**issue(non-blocking)**: This is potentially A LOT of pickings that you're loading in memory.

I marked it as non-blocking just because it's not technically part of thsis PR, but since the module is still new, I think it's a good time to fix it.

Also, a few lines below, you then only want to operate on pickings that have the `intercompany_picking_id` field set, so you could already add that condition here. Ideally, though, I think you'd want to do a `search` in the `for` loop instead of loading all possible pickings here.

Or just use `note.picking_ids`.

Reply:
Thanks for the beautiful suggestion, I'm still not totally sure why in the past I wrote that superficial `filtered`, but using `note.picking_ids` totally works out.